plugins/modules/mso_schema_template_anp_epg_useg_attribute.py

```diff
@@ -0,0 +1,269 @@
+        schema=dict(type="str"),
+        template=dict(type="str"),
+        anp=dict(type="str"),
+        epg=dict(type="str"),
+        name=dict(type="str", aliases=["useg"]),
+        description=dict(type="str", aliases=["descr"]),
+        attribute_type=dict(type="str", aliases=["attr_type"], choices=list(EPG_U_SEG_ATTR_TYPE_MAP.keys())),
+        value=dict(type="str"),
+        operator=dict(type="str", choices=["equals", "contains", "starts_with", "ends_with"]),
+        useg_subnet=dict(type="bool"),
+        state=dict(type="str", default="present", choices=["absent", "present", "query"]),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ["state", "query", ["schema", "template", "anp", "epg"]],
+            ["state", "absent", ["schema", "template", "anp", "epg", "name"]],
+            ["state", "present", ["schema", "template", "anp", "epg", "name", "attribute_type"]],
+            ["useg_subnet", False, ["value"]],
+        ],
+    )
+
+    schema = module.params.get("schema")
+    template = module.params.get("template").replace(" ", "")
+    anp = module.params.get("anp")
+    epg = module.params.get("epg")
+
+    name = module.params.get("name")
+    description = module.params.get("description")
+
+    attribute_type = module.params.get("attribute_type")
+    value = module.params.get("value")
+    operator = module.params.get("operator")
+
+    useg_subnet = module.params.get("useg_subnet")
+
+    state = module.params.get("state")
+
+    mso = MSOModule(module)
+    if state == "present":
+        if attribute_type in ["mac", "dns"] and value is None:
+            mso.fail_json(msg="Failed due to invalid 'value' and the attribute_type is: {0}.".format(attribute_type))
+        elif attribute_type in [
+            "vm",
+            "vm_datacenter",
+            "vm_hypervisor_identifier",
+            "vm_operating_system",
+            "vm_tag",
+            "vm_identifier",
+            "vmm_domain",
+            "vnic_dn",
+            "vm_name",
+        ] and (value is None or operator is None):
+            mso.fail_json(msg="Failed due to invalid 'value' or 'operator' and the attribute_type is: {0}.".format(attribute_type))
+
+        if attribute_type == "ip" and useg_subnet is False:
+            fvSubnet = True
+        elif attribute_type == "ip" and useg_subnet is True:
+            value = "0.0.0.0"
+            fvSubnet = False
+
+    mso_schema = MSOSchema(mso, schema, template)
+    mso_schema.set_template(template, fail_module=True)
+    mso_schema.set_template_anp(anp, fail_module=True)
+    mso_schema.set_template_anp_epg(epg, fail_module=True)
+
+    if mso_schema.schema_objects["template_anp_epg"].details.get("uSegEpg"):
+        match_useg_attr, useg_attrs = mso_schema.get_object_from_list(
+            mso_schema.schema_objects["template_anp_epg"].details.get("uSegAttrs"), [KVPair("name", name)]
```

Comment:
Is uSegAttrs always a list also when it is empty? If not we probably need to add check or change the get_object_from_list() function to handle None values.

Reply:
uSegAttrs - will be always a list.